File: 6_kyu/Make_the_Deadfish_swim.py
def parse(data):
    value = 0
    output = []
    for command in data:
        if command == "i":
            value += 1
        elif command == "d":
            value -= 1
        elif command == "s":
            value *= value
        elif command == "o":
            output.append(value)
        # ignore invalid command
    return output

File: 6_kyu/Make_the_Deadfish_Swim.go
package kata

func Parse(data string) []int{
    var value int
    output := make([]int,0)
    for _,command := range data {
        switch command {
        case 'i' :
            value++
        case 'd' :
            value--
        case 's' :
            value *= value
        case 'o' :
            output = append(output,value)
        }
    }
    return output
}